src/phoenix/server/api/routers/v1/datasets.py

```diff
@@ -450,3 +516,100 @@ async def _parse_form_data(
         metadata_keys,
         file,
     )
+
+
+async def _get_dataset_download_csv(request: Request) -> Response:
+    encoded_dataset_id = request.path_params.get("datasetId")
+    if not encoded_dataset_id:
+        return Response(
+            content="Missing Dataset ID",
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+        )
+    dataset_global_id = GlobalID.from_id(encoded_dataset_id)
+    if dataset_global_id.type_name != Dataset.__name__:
+        return Response(
+            content=f"Invalid Dataset ID: {encoded_dataset_id}",
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+        )
+    encoded_dataset_version_id = request.path_params.get("datasetVersionId")
+    dataset_version_global_id = (
+        GlobalID.from_id(encoded_dataset_version_id) if encoded_dataset_version_id else None
+    )
+    if dataset_version_global_id and dataset_version_global_id.type_name != DatasetVersion.__name__:
+        return Response(
+            content=f"Invalid Dataset Version ID: {encoded_dataset_version_id}",
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+        )
+    dataset_id = int(dataset_global_id.node_id)
+    mder = models.DatasetExampleRevision
+    latest_version = (
+        select(
+            mder.dataset_example_id,
+            (func.rank().over(order_by=mder.dataset_example_id) - 1).label("example_index"),
+            func.max(mder.dataset_version_id).label("dataset_version_id"),
+        )
+        .group_by(mder.dataset_example_id)
+        .join(models.DatasetExample)
+        .where(models.DatasetExample.dataset_id == dataset_id)
+    )
+    if dataset_version_global_id:
+        max_dataset_version_id = int(dataset_version_global_id.node_id)
+        latest_version = latest_version.where(mder.dataset_version_id <= max_dataset_version_id)
+    subq = latest_version.subquery("latest_version")
+    stmt = (
+        select(
+            mder.input,
+            mder.output,
+            mder.metadata_,
+            subq.c.example_index,
+        )
+        .join(
+            subq,
+            onclause=and_(
+                mder.dataset_example_id == subq.c.dataset_example_id,
+                mder.dataset_version_id == subq.c.dataset_version_id,
+            ),
+        )
+        .where(mder.revision_kind != "DELETE")
+        .order_by(mder.dataset_example_id)
+    )
+    async with request.app.state.db() as session:
+        dataset_name: Optional[str] = await session.scalar(
+            select(models.Dataset.name).where(models.Dataset.id == dataset_id)
+        )
+        if not dataset_name:
+            return Response(
+                content="Dataset does not exist.",
```

Comment:
Suggestion: Return the dataset name here.

Reply:
`dataset_name` is missing/empty here

Reply:
Whoops lol